cloudmarker/stores/mongodbstore.py

```diff
@@ -3,51 +3,46 @@
 
 import logging
 
-from pymongo import MongoClient
+from pymongo import MongoClient, errors
 
 _log = logging.getLogger(__name__)
 
 
 class MongoDBStore:
     """A plugin to store records on MongoDB."""
 
-    def __init__(self, host, port, username=None, password=None, db='gcp',
-                 buffer_size=1000):
+    def __init__(self, db, username, password, host, port=27017,
+                 buffer_size=1000, **kwargs):
         """Create an instance of :class:`MongoDBStore` plugin.
 
-        Arguments:
-            host (str): MongoDB instance host.
-            port (int): MongoDB instance port.
-            username (str): MongoDB instance auth username.
-            password (str): MongoDB instance auth password.
-            db (str): MongoDB database name, default: ``gcp``
-            buffer_size (int): Max number of records to hold in memeory for
-                each record_type.
+        It will use the default port for mongodb 27017 if not specified.
+        The Authentication scheme will be negotiated by MongoDB and the client
+        for v4.0+ to SCRAM-SHA-1 or SCRAM-SHA-256 by default aftere
+        negotiation.
 
+        Arguments:
+            db (str): name of the database
+            username (str): username for the database
+            password (str): password for username to authenticate with the db
+            host (str): hostname for the DB server
+            port (int): port for mongoDB is listening, defaults to 27017
+            buffer_size (int): max buffer before flushing to db
+            kwargs (dict): Additional args
+                * models - List of classes for validator and enforcement
+                    requirements.
         """
-        # pylint: disable=too-many-instance-attributes
```

Comment:
Removal of `pylint` directive has to be a separate commit. Every commit should do exactly what the commit summary line says and no more. Further, the commit summary line cannot be generic phrases "code cleanup" or "error handling". 🙂

This single line change should be a separate commit (could be a separate pull request or part of any pull request) that clearly tells in the summary line, for example, `Remove Pylint control directive from MongoDBStore`. Worth noting in this example:

- The summary line is very specific about what change is made.
- The brand names "Pylint" is capitalized just the way the Pylint community has named their project.

Then the commit message should elaborate why it is okay to remove this `Pylint` directive. I believe the answer is that `pylama.ini` already has the necessary `ignore` keys which makes this directive redundant.